### PR TITLE
[feature](cloud-compaction) Support shadow tablet to do cumulative compaction during schema change in cloud mode

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

In cloud mode, when do schema change, shadow tablet encounters -235 because it cant do cumulative compaction in the case of a large number of loads. And it will prevents the user from continuing to loads. Implementation details:
1. When start schema change, record the end convert rowset version `alter_version` into SchemaChangeJob.
2. For origin tablet, only can do base compaction in [0, `alter_version`] and do cumulative compaction in (`alter_version`, N]. can not do compaction across `alter_verison` such as compaction [a, `alter_version` + n].
3. For shadow tablet, cannot do base compaction and and do cumulative compaction in (`alter_version`, N].
4. When the schema change failed because FE or BE coredump, it will retry. When retry the schema change, it will get the `alter_version` from meta_serive, and continue to do it.
5. When finish the schema change job or cancel it, we need to clear the schema change job. Before this pr, it will cover by next schema change.

co-author(main author): @Lchangliang 
original PR: #37293

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run cloud_p0

**Comment by TangSiyang2001** [comment]
run cloud_p0

**Comment by TangSiyang2001** [comment]
run cloud_p0

File: cloud/src/meta-service/meta_service_job.cpp [hunk 1]
```diff
@@ -124,6 +150,17 @@ void start_compaction_job(MetaServiceCode& code, std::string& msg, std::stringst
     }
     while (err == TxnErrorCode::TXN_OK) {
         job_pb.ParseFromString(job_val);
+        if (!check_compaction_input_verions(compaction, job_pb)) {
+            SS << "Check compaction input versions failed in schema change. input_version_start="
+               << compaction.input_versions(0)
+               << " input_version_end=" << compaction.input_versions(1)
```
Comment (TangSiyang2001) [inline cloud/src/meta-service/meta_service_job.cpp:156]: Here is supicious

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run external

**Comment by TangSiyang2001** [comment]
run cloud_p0

**Review by gavinchou — DISMISSED** [review]

**Comment by TangSiyang2001** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by luwei16 — APPROVED** [review]

**Pull request merged**